upup/pkg/fi/fitasks/managedfile.go

```diff
@@ -35,10 +36,12 @@ type ManagedFile struct {
 	Name      *string
 	Lifecycle fi.Lifecycle
 
-	Base     *string
-	Location *string
-	Contents fi.Resource
-	Public   *bool
+	Base            *string
+	Location        *string
+	Contents        fi.Resource
+	Public          *bool
+	hash            *string
+	renderedContent *string
```

Comment:
This appears to be a redundant copy of `Contents`, converted to `string`. Why is it necessary?

Reply:
As you maybe saw, it is because rendering the resource in this case can be worth caching. But as you also commented, this memoization can/should be done in the resource itself.